### PR TITLE
Sign compare warning fixes batch 2

### PR DESCRIPTION
**Pull request by tg-at-google**
In resolution of warnings with ids: [
47, 48, 49, 51, 52,
53, 54, 55, 56, 57, 
59, 60, 61, 63, 65, 70, 
]

@mihaimaruseac

### CODE REVIEW

File: tensorflow/core/grappler/costs/graph_properties.cc [hunk 1]
```diff
@@ -1357,18 +1362,20 @@ class SymbolicShapeRefiner {
   // Returns true if all the output tensors have known values.
   bool AllOutputValuesKnown(NodeContext* c) {
     InferenceContext* ic = c->inference_context.get();
-    if (c->output_tensors_as_shapes.size() < ic->num_outputs() &&
-        c->output_tensor_protos.size() < ic->num_outputs()) {
+    int c_output_tensors_as_shapes_size = c->output_tensors_as_shapes.size();
+    int c_output_tensor_protos_size = c->output_tensor_protos.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/costs/graph_properties.cc:1366]: I would move these two on the `else` block to only be defined if that branch is taken. Eventually even move them inside the `for` there.

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/core/kernels/data/experimental/sql/sqlite_query_connection.cc [hunk 2]
```diff
@@ -68,7 +68,7 @@ Status SqliteQueryConnection::GetNext(IteratorContext* ctx,
 
 Status SqliteQueryConnection::PrepareQuery() {
   TF_RETURN_IF_ERROR(db_->Prepare(query_, &stmt_));
-  int column_count = stmt_.ColumnCount();
+  size_t column_count = stmt_.ColumnCount();
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/data/experimental/sql/sqlite_query_connection.cc:71]:
```
tensorflow/core/kernels/data/experimental/sql/sqlite_query_connection.cc:77:9: error: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Werror,-Wformat]
        column_count, output_types_.size()));
```

Let's keep this int and static cast the `output_types_.size()` instead

File: tensorflow/core/grappler/costs/graph_properties.cc (hunk 1 above)
Comment (tg-at-google) [reply tensorflow/core/grappler/costs/graph_properties.cc:1366]: They are actually used in before the `else` block, in the `if`.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by tg-at-google** [comment]
@joker-eph  There are no mlir related changes in this PR.

**Pull request merged**